openmdao/core/total_jac.py

```diff
@@ -17,23 +17,27 @@
 from openmdao.utils.mpi import MPI, multi_proc_exception_check
 from openmdao.utils.coloring import _initialize_model_approx, Coloring
 
-# Attempt to import petsc4py.
-# If OPENMDAO_REQUIRE_MPI is set to a recognized positive value, attempt import
-# and raise exception on failure. If set to anything else, no import is attempted.
+
 if 'OPENMDAO_REQUIRE_MPI' in os.environ:
-    if os.environ['OPENMDAO_REQUIRE_MPI'].lower() in ['always', '1', 'true', 'yes']:
-        from petsc4py import PETSc
+    # If OPENMDAO_REQUIRE_MPI is set to a recognized positive value, import
+    # petsc4py and raise ImportError on failure.
+    if os.environ['OPENMDAO_REQUIRE_MPI'].lower() in ['always', '1', 'true', 'yes', 'y', 'on']:
+        try:
+            from petsc4py import PETSc
+
+        except ImportError:
+            raise ImportError("Importing petsc4py failed and OPENMDAO_REQUIRE_MPI is true.")
     else:
+        # OPENMDAO_REQUIRE_MPI is set to something else, so no import is attempted.
```

Comment:
It seems like `OPENMDAO_REQUIRE_MPI` means it's okay either way, as opposed to "I don't want MPI"

I would expect the test to be:

> if `OPENMDAO_REQUIRE_MPI` exists and is truthy, then failure to import MPI/PETSc is an error, otherwise try and fail silently

Reply:
I wanted to keep an option to prevent the attempt, in case it _is_ installed but you don't want to load it for some reason. Maybe it's broken (can happen after updates) or you just want to be confident that you're doing a serial run.

Reply:
I guess then it's the var name that just doesn't convey the meaning to me..   
This sounds more like `OPENMDAO_USE_MPI`  
(True to use it and fail if you can't, False to not even try)

Reply:
Agree, but the problem with changing the var name at this point is that `OPENMDAO_REQUIRE_MPI` has been around for almost 2 years. The only difference with this PR is what happens when it's not set.